### ISSUE TITLE
Setting default params 

### ISSUE THREAD

**Issue by Sokal21** [axios/axios#1476]
#### Summary

Is there a way to set default params to all type of request methods? I check this issue https://github.com/axios/axios/issues/229 but the solution that was provided do not work for me. I'm doing something bad or this work around is not available anymore?

#### Context

- axios version: 0.18.0
- Environment: *node v8.11.1, Opera 52, windows 10*

**Comment by baumannzone** [comment]
You should show your code

**Comment by melbahja** [comment]
Try
```js
axios.defaults.params = 
{
   name: 'value'
}
```

**Comment by pk936** [comment]
default params are not working for me too.

**Comment by libasoles** [comment]
Not working here neither:

`const api = axios.create({
  baseURL: config.api.baseURL,
  params: {
    appid: 'mysecretkey',
  },
});`

Also tried:
`axios.defaults.params = {
  appid: 'mysecretkey',
};`

And I must say I'm not overriding the params when performing a query. So, they should be there according to the docs.

**Comment by libasoles** [comment]
Issue, they say, is with v0.19. Downgrading to 0.18.1 should work. See:
#2190

There's a working demo of this:
https://runkit.com/philipbjorge/axios-regression

However, I downgraded and still can make it work.

**Comment by pk936** [comment]
@libasoles  thanks.. downgraded to 0.18.1 and it works !

**Comment by gualcm** [comment]
Downgrading to 0.18.1 doesn't fix the issue anymore.

**Comment by yassinechabli** [comment]
Having the some issue with axios 0.19.0 . downgrading to 0.18.1 resolve the problem . 
@libasoles +1

**Comment by fsgiudice** [comment]
same problem here, downgraded to 0.18.1 and it works. Thank you

**Comment by horpto** [comment]
I have the same problem. solved with interceptors:

```
const ax = axios.create({
    headers: {
        'X-Requested-With': 'XMLHttpRequest'
    },
});
ax.interceptors.request.use((config) => {
    config.params = config.params || {};
    config.params['token'] = token;
    return config;
});
```

**Comment by zalog** [comment]
Same here, not working.
Tnks @horpto +1.

**Comment by chinesedfan** [comment]
Closed in favor of #2190.
